homeassistant/components/onewire/config_flow.py

```diff
@@ -164,3 +177,163 @@ async def async_step_mount_dir(
+    """Handle OneWire Config options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize OneWire Network options flow."""
+        self.entry_id = config_entry.entry_id
+        self.options = dict(config_entry.options)
+        self.configurable_devices: dict[str, OWServerDeviceDescription] = {}
+        self.devices_to_configure: dict[str, OWServerDeviceDescription] = {}
+        self.current_device: str = ""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options."""
+        controller: OneWireHub = self.hass.data[DOMAIN][self.entry_id]
+        if controller.type == CONF_TYPE_SYSBUS:
+            return self.async_abort(
+                reason="SysBus setup does not have any config options."
+            )
+
+        all_devices: list[OWServerDeviceDescription] = controller.devices  # type: ignore[assignment]
+        if not all_devices:
+            return self.async_abort(reason="No configurable devices found.")
+
+        device_registry = dr.async_get(self.hass)
+        self.configurable_devices = {
+            self._get_device_long_name(device_registry, device.id): device
+            for device in all_devices
+            if device.family in DEVICE_SUPPORT_OPTIONS
+        }
+
+        return await self.async_step_device_selection(user_input=None)
+
+    async def async_step_device_selection(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Select what devices to configure."""
+        errors = {}
+        if user_input is not None:
+            if user_input.get(INPUT_ENTRY_CLEAR_OPTIONS):
+                # Reset all options
+                self.options = {}
+                return await self._update_options()
+
+            selected_devices: list[str] = (
+                user_input.get(INPUT_ENTRY_DEVICE_SELECTION) or []
+            )
+            if selected_devices:
+                self.devices_to_configure = {
+                    device_name: self.configurable_devices[device_name]
+                    for device_name in selected_devices
+                }
+
+                return await self.async_step_configure_device(user_input=None)
+            errors["base"] = "device_not_selected"
+
+        return self.async_show_form(
+            step_id="device_selection",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        INPUT_ENTRY_CLEAR_OPTIONS,
+                        default=False,
+                    ): bool,
+                    vol.Optional(
+                        INPUT_ENTRY_DEVICE_SELECTION,
+                        default=self._get_current_configured_sensors(),
+                        description="Multiselect with list of devices to choose from",
+                    ): cv.multi_select(
+                        {device: False for device in self.configurable_devices.keys()}
```

Comment:
We can remove `.keys()`. It's the default dict iterator.

Reply:
OK